probe.go

```diff
@@ -707,30 +730,46 @@ func (p *Probe) attachUprobe() error {
 		p.HookFuncName = offsets[0].Name
 	}
 
-	// enable uprobe
-	uprobeID, err := EnableUprobeEvent(p.GetUprobeType(), p.HookFuncName, p.BinaryPath, p.UID, p.attachPID, p.UprobeOffset)
+	// Try to use the perf_event_open API first (e12f03d "perf/core: Implement the 'perf_kprobe' PMU")
+	p.perfEventFD, err = perfEventOpenPMU(p.BinaryPath, int(p.UprobeOffset), -1, "uprobe", p.GetUprobeType() == "r", 0)
```

Comment:
Hey Bryce,
I took into account your comment from last time, and if you look at the code, the first thing that `perfEventOpenPMU` does is call `getPMUEventType` which has a `sync.One` to only make one syscall. In other words, if the PMU abi is not supported, it will make one syscall only the first time and never make it again: it will immediately fall back to the kprobe_events method.

Reply:
Ah, I missed that. Great!